Ukoly/DU_10/generator.py

```diff
@@ -0,0 +1,118 @@
+#
+# české názvy jsou ponechány po vzoru zadání
+
+from datetime import datetime, timedelta
+import requests
+import random
+import json
+import qrcode
+now = datetime.now()
+url = (f'https://data.kurzy.cz/json/meny/b[2]den[{now.strftime("%Y%m%d")}].json')
+
+
+class CurrencyError(Exception):
+    def __init__(self, name):
+        self.name = name
+
+    def __str__(self):
+        return(f'Tato měna není na seznamu')
+
+
+class LoadExchangeRates():
+    def __init__(self, name):
+        self.name = name
+
+    def load_data(self):
+        exchange = requests.get(url=url)
+        with open("kurz_cnb.json", 'wb') as text_file:
+            text_file.write(exchange.content)
+
+    def exchange_rates(self):
+        with open("kurz_cnb.json", 'r') as text_file:
+            data = json.load(text_file)
+            json_date = datetime.strptime(data["den"], "%Y%m%d")
+            if now - json_date > timedelta(days=1):
+                LoadExchangeRates.load_data(self)
+
+
+class Payment:
+    def __init__(self, name, currency, amount_of_money):
+        self.name = name
+        self.currency = currency
+        self.amount_of_money = amount_of_money
+
+    def run_app(self):
+        if self.currency != 'KČ' or self.currency == None:
+            Convertor.convert(self)
+        LoadExchangeRates.load_data(self)
+        Generator.vs_generator(self)
+        Generator.qr_generator(self)
+
+
+class Convertor(Payment):
+    def convert(self):
+        with open("kurz_cnb.json", "r") as text_file:
+            data = json.load(text_file)
+            if self.currency in data["kurzy"]:
+                exchange_rate = data["kurzy"][self.currency]["dev_stred"]
+                exchange_unit = 1/data["kurzy"][self.currency]["jednotka"]
+                self.amount_of_money = int(self.amount_of_money) *\
+                    exchange_unit * exchange_rate
+                type(exchange_unit)
+            else:
```

Comment:
Ak by si celú túto podmeinku otočila, tak by si sa zbavila `else` vetvy (nakoľko výnimku nikde v tejto funkcii neodchytávaš, správala by sa dosť ekvivalentne `returnu`)